compiler/parser/src/error.rs

```diff
@@ -220,7 +220,7 @@ impl From<FStringError> for LalrpopError<Location, Tok, LexicalError> {
 pub type ParseError = rustpython_compiler_core::BaseError<ParseErrorType>;
 
 /// Represents the different types of errors that can occur during parsing.
-#[derive(Debug, PartialEq, thiserror::Error)]
+#[derive(Debug, PartialEq)]
```

Comment:
thiserror is fine, except that because we depend on this from `rustpython-derive`, this would be a proc-macro depending on a proc-macro. With this PR, `rustpython/derive $ cargo tree -i syn` returns no results. We're kind of an outlier in doing so much inside of a proc-macro, so I want to try to make the impact on compile times as small as possible.